packages/core/src/render3/instructions.ts

```diff
@@ -374,9 +435,9 @@ function getTemplateStatic(template: ComponentTemplate<any>): NgStaticData {
 
 function setUpAttributes(native: RElement, attrs: string[]): void {
   ngDevMode && assertEqual(attrs.length % 2, 0, 'attrs.length % 2');
-  const isFnRenderer = (renderer as Renderer3Fn).setAttribute;
+  const isFnRenderer = (renderer as ProceduralRenderer3).setAttribute;
```

Comment:
`isFnRenderer` -> `isProceduralRenderer` throughout? 

Actually, could this be a module-level var that's computed once when `renderer` is set? Or, even shorter, have a module-level var `proceduralRenderer` that starts as `null` and is only assigned if it's the right type, which would let the rest of the code just be
```ts
proceduralRenderer ? 
    proceduralRenderer.doSomething(...) :
    (renderer as ObjectOrientedRenderer).doSomething(...);
```